examples/assets/scripts/animation/tween.js

```diff
@@ -0,0 +1,318 @@
+            type: 'string'
+        }, {
+            name: 'stopEvent',
+            title: 'Stop Event',
+            description: 'Executed when a tween is explicitly stopped via stop(), but not when it is completed normally.',
+            type: 'string'
+        }, {
+            name: 'updateEvent',
+            title: 'Update Event',
+            description: 'Executed each time the tween is updated, after the values have been actually updated.',
+            type: 'string'
+        }, {
+            name: 'completeEvent',
+            title: 'Complete Event',
+            description: 'Executed when a tween is finished normally (i.e. not stopped).',
+            type: 'string'
+        }, {
+            name: 'repeatEvent',
+            title: 'Repeat Event',
+            description: 'Executed whenever a tween has just finished one repetition and will begin another.',
+            type: 'string'
+        }
+    ],
+    array: true
+});
+
+// initialize code called once per entity
+Tween.prototype.initialize = function () {
+    var app = this.app;
+    var i;
+
+    this.tweenInstances = [];
+    this.tweenCallbacks = [];
+
+    var makeStartCallback = function (i) {
+        return function () {
+            this.start(i);
+        };
+    };
+
+    for (i = 0; i < this.tweens.length; i++) {
+        var tween = this.tweens[i];
+        if (tween.autoPlay) {
+            this.start(i);
+        }
+        if (tween.event && tween.event.length > 0) {
+            this.tweenCallbacks[i] = {
+                event: tween.event,
+                cb: makeStartCallback(i)
+            };
+            app.on(this.tweenCallbacks[i].event, this.tweenCallbacks[i].cb, this);
+        }
+    }
+
+    // Resume all paused tweens if the script is enabled
+    this.on('enable', function () {
+        for (i = 0; i < this.tweens.length; i++) {
+            if (this.tweenInstances[i] && this.tweenInstances[i].isPaused()) {
+                if (this.tweenInstances[i].isPaused()) {
+                    this.tweenInstances[i].resume();
+                }
+            }
+        }
+    });
+
+    // Pause all playing tweens if the script is disabled
+    this.on('disable', function () {
+        for (i = 0; i < this.tweens.length; i++) {
+            if (this.tweenInstances[i]) {
+                if (this.tweenInstances[i].isPlaying()) {
+                    this.tweenInstances[i].pause();
+                }
+            }
+        }
+    });
+
+    this.on('attr', function (name, value, prev) {
+        for (i = 0; i < this.tweenCallbacks.length; i++) {
+            if (this.tweenCallbacks[i]) {
+                app.off(this.tweenCallbacks[i].event, this.tweenCallbacks[i].cb, this);
+                this.tweenCallbacks[i] = null;
+            }
+        }
+
+        for (i = 0; i < this.tweens.length; i++) {
+            var tween = this.tweens[i];
+            if (tween.event.length > 0) {
```

Comment:
no autoplay here?

Reply:
No, because we're just rewiring the (potentially changed) events here. Even if `autoPlay` on a tween was flipped from `false` to `true`, I'm not sure the tween should be started. Autoplay should occur on script initialization alone IMHO.